Add How To for global gitleaks pre-commit check (#131)
giantswarm/fmt: go/general_style_guide.md

Please don't push secrets to GitHub unless they are encrypted.

We recommend [configuring a client-side git hook](https://github.com/zricethezav/gitleaks/wiki/Scanning#uncommitted-changes-scan) so that your secrets are never pushed in the first place. To do this:
```
# Create a directory to hold your global git hooks, if you don't have one
$ mkdir /path/to/your/git-hooks

> # Create a file named pre-commit in that directory, with the content from the gitleaks link above

Reviewer: Can we add that in here directly? shouldn't be too hard to `touch` a file and then pipe that stuff in

Author: Yeah I can include it here, my thinking was that this might get out of date with upstream then

Please don't push secrets to GitHub unless they are encrypted.

We recommend [configuring a client-side git hook](https://github.com/zricethezav/gitleaks/wiki/Scanning#uncommitted-changes-scan) so that your secrets are never pushed in the first place. To do this:

> ```

Reviewer 2: ```suggestion
```bash
```

# Create a directory to hold your global git hooks, if you don't have one
$ mkdir /path/to/your/git-hooks

# Change to the directory
$ cd /path/to/your/git-hooks

# Create a file named pre-commit
$ touch pre-commit

# Add the git hook content from the gitleaks wiki
$ cat <<EOS >> pre-commit
#!/bin/sh
# Adds gitleaks as a pre-commit hook.

gitleaksEnabled=\$(git config --bool hooks.gitleaks)
cmd="gitleaks --verbose --redact --pretty"

> if [ "\$gitleaksEnabled" = "true" ]; then

Reviewer 2: I don't think it will work if you don't run `git config hooks.gitleaks true` first

```suggestion
if [ "\$gitleaksEnabled" != "false" ]; then
```

Author: If the user hasn't set it then the value is just an empty string, not false

    \$cmd
    if [ \$? -eq 1 ]; then

> cat <<\EOF
> Error: gitleaks has detected sensitive information in your changes.
> If you know what you are doing you can disable this check using:
>     git config hooks.gitleaks false
> EOF
> exit 1

Reviewer 2: To make it look (hopefully) less awful:

```suggestion
        echo >&2 "Error: gitleaks has detected sensitive information in your changes."
        echo >&2 "If you know what you are doing you can disable this check using:"
        echo >&2 "      git config hooks.gitleaks false"
        exit 1
```

Author: I'd like to keep the script the same as the upstream one just to reduce the overhead of keeping it updated

Reviewer 2: I didn't know it's upstream script. Proceed as you see the fit.
